cmake/GhcHelper.cmake

```diff
@@ -22,6 +22,9 @@ if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND (CMAKE_CXX_COMPILER_VERSION V
     if(${CMAKE_SYSTEM_NAME} MATCHES "(SunOS|Solaris)")
         target_link_libraries(filesystem_test xnet)
     endif()
+    if(${CMAKE_SYSTEM_NAME} MATCHES "(Haiku)")
```

Comment:
The parentheses around Haiku are unnecessary (there aren't multiple alternatives here)